compiler/semtypes.nim

```diff
@@ -1578,6 +1578,13 @@ proc semGeneric(c: PContext, n: PNode, s: PSym, prev: PType): PType =
         localError(c.config, n.info, errCannotInstantiateX % s.name.s)
         result = newOrPrevType(tyError, prev, c)
       else:
+        #fixes 1500
+        if n[0].kind == nkSym and n[0].sym.ast != nil and n[0].sym.ast.len > 1 and n[0].sym.ast[2].kind == nkObjectTy:
```

Comment:
Turn this whole check into a helper proc please. Name it `containsGenericInvocationWithForward`